Applications/DataExplorer/DataView/MeshElementRemovalDialog.cpp

```diff
@@ -202,24 +262,48 @@ void MeshElementRemovalDialog::on_meshNameComboBox_currentIndexChanged(int idx)
     this->_currentIndex = this->meshNameComboBox->currentIndex();
     this->newMeshNameEdit->setText(this->meshNameComboBox->currentText() + "_new");
     this->elementTypeListWidget->clearSelection();
-    this->materialListWidget->clearSelection();
-    if (this->boundingBoxCheckBox->isChecked()) this->on_boundingBoxCheckBox_toggled(true);
-    auto mesh = _project.getMesh(meshNameComboBox->currentText().toStdString());
-    if (mesh->getProperties().existsPropertyVector<int>("MaterialIDs"))
-    {
-        auto const* const materialIds =
-            mesh->getProperties().getPropertyVector<int>("MaterialIDs");
-        if (materialIds->size() != mesh->getNumberOfElements())
-        {
-            ERR ("Incorrect mesh structure: Number of Material IDs does not match number of mesh elements.");
-            OGSError::box("Incorrect mesh structure detected.\n Number of Material IDs does not match number of mesh elements");
-            QMetaObject::invokeMethod(this, "close", Qt::QueuedConnection);
-        }
+    this->scalarArrayComboBox->clear();
+    this->outsideScalarMinEdit->setText("");
+    this->outsideScalarMaxEdit->setText("");
+    this->insideScalarMinEdit->setText("");
+    this->insideScalarMaxEdit->setText("");
+    if (this->scalarArrayCheckBox->isChecked())
+        on_scalarArrayCheckBox_toggled(true);
+    if (this->boundingBoxCheckBox->isChecked())
+        on_boundingBoxCheckBox_toggled(true);
+}
+
+void MeshElementRemovalDialog::on_scalarArrayComboBox_currentIndexChanged(int idx)
+{
+    Q_UNUSED(idx);
+    MeshLib::Mesh const* const mesh =
+        _project.getMesh(meshNameComboBox->currentText().toStdString());
+    MeshLib::Properties const& properties = mesh->getProperties();
```

Comment:
The mesh pointer is not checked before access, could be nullptr.

Reply:
Added the nullptr-check (although if there was a problem the programme would've crashed long before this point), but I'm not sure what you mean with your second remark. 

The ```if```` applies if the currently selected mesh has no scalar arrays. In this case, the combobox will be empty and therefore the returned string will be empty. No range values can be computed and all related widgets will remain disabled.

If the has *does* have scalar arrays, ```vec_name``` cannot be empty, thus range values are computed, widgets are enabled, etc.

Does that answer your comment?

Reply:
The second remark is only for the order of the lines; Should be
```c++
if (vec_name.empty()) return;
auto mesh = getMesh();
auto properties = getProperties(mesh);
```
The `properties` are not needed in case the `vec_name` is empty...

Reply:
I see. Fixed! :heavy_check_mark: